lib/elixir/lib/calendar.ex

```diff
@@ -1862,11 +1911,76 @@ defmodule DateTime do
       :gt
   """
   @spec compare(DateTime.t, DateTime.t) :: :lt | :eq | :gt
-  def compare(%DateTime{} = datetime1, %DateTime{} = datetime2) do
-    case {to_unix(datetime1, :microsecond), to_unix(datetime2, :microsecond)} do
+  def compare(%DateTime{calendar: calendar1} = datetime1, %DateTime{calendar: calendar2} = datetime2) do
+    {year1, month1, day1, hour1, minute1, second1, microsecond1, time_zone1, zone_abbr1, utc_offset1, std_offset1} = datetime1 |> to_tuple
```

Comment:
I don't understand why you are calling `to_tuple` and then matching on the tuple. :) If you want to avoid duplication, it would be better to encapsulate the pattern matching and the call to datetime_to_rata_die instead?